src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  Button,
+  Flex,
+  LabeledInput,
+  ModalDialog,
+  Spacer,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import EditIcon from "@artsy/icons/EditIcon"
+import { useTranslation } from "react-i18next"
+import { SavesArtworks_collection$data } from "__generated__/SavesArtworks_collection.graphql"
+import { Form, Formik, FormikHelpers } from "formik"
+import * as Yup from "yup"
+import { pick } from "lodash"
+import { useUpdateCollection } from "./Mutations/useUpdateCollection"
+import createLogger from "Utils/logger"
+
+interface EditSavesModalProps {
+  collection: SavesArtworks_collection$data
+  setIsEditModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+interface FormikValues {
+  name: string
+}
+
+const MAX_NAME_LENGTH = 40
+
+const logger = createLogger("EditSavesModal")
+
+export const EditSavesModal: React.FC<EditSavesModalProps> = ({
+  collection,
+  setIsEditModalOpen,
+}) => {
+  const { t } = useTranslation()
+
+  const initialValues = pick(collection, "name")
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .required(t("collectorSaves.editListModal.fields.name.required")) // TODO: confirm copy
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useUpdateCollection()
+
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async (
+    formikValues: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>
+  ) => {
+    try {
+      await submitMutation({
+        variables: {
+          input: {
+            id: collection.internalID,
+            name: formikValues.name,
+          },
+        },
+        rejectIf: res => {
+          const result = res.updateCollection?.responseOrError
+
+          return result?.__typename === "UpdateCollectionFailure"
+            ? result.mutationError
+            : false
+        },
```

Comment:
I see two problems with that approach:

- The smaller problem is that the above code doesn't pass TS, without some more type narrowing perhaps

- The bigger problem is that this would return a boolean instead of the actual error message, and thus swallows a potentially helpful user-facing error message

|With `!!responseOrError.mutationError` — <br/>generic error message|With `responseOrError.result.mutationError` — <br/>actual error message|
|---|---|
|<img width="100%" alt="bool" src="https://user-images.githubusercontent.com/140521/222744845-0e2af033-145c-424c-b2e7-383463afd616.png">|<img width="100%" alt="msg" src="https://user-images.githubusercontent.com/140521/222745371-c3aad266-1c77-4604-a62e-47be04e23af0.png">|

I actually think we should switch to _this_ error-handling pattern more broadly, so that we don't hide useful error messages.